File: tests/unitTests/customEvent.lua
local ServerStorage = game:GetService("ServerStorage")

local DatalinkService = require(ServerStorage.DatalinkService)

local REQUEST_SIZE = 0

local customEventTest = { }

function customEventTest.countDownFrom(x)
	for index = 1, x do
		print(script.Name, x - index)

		task.wait(1)
	end
end

function customEventTest.run()
	for index = 1, REQUEST_SIZE do
		task.spawn(function()
			DatalinkService:FireCustomEvent("EventExample", "EventData1", { })
		end)
	end

	return true
end

return customEventTest

File: src/tests/ts/unitTests/customEvent.ts
import DatalinkService from '../../..'

const REQUEST_SIZE = 0

class customEventTest {
    run() {
        for (let index = 1; index < REQUEST_SIZE; index++) {
            task.spawn(() => {
                return DatalinkService.FireCustomEvent("EventExample", "EventData1", { }).Then((index: number) => {
                    print("Completed: ", index)
                })
            })
        }

    }
}

export { customEventTest }